thirdeye/thirdeye-frontend/app/pods/rootcause-aggregates-cache/service.js

```diff
@@ -62,11 +62,37 @@ export default Service.extend({
       offsetsAndUrns.push([toAbsoluteUrn(urn, requestContext.compareMode).split(':')[2].toLowerCase(), urn]);
       metricUrnToOffestAndUrn[metricUrn] = offsetsAndUrns;
     });
-
-    Object.keys(metricUrnToOffestAndUrn).forEach(
+    const metricUrns = Object.keys(metricUrnToOffestAndUrn);
+    metricUrns.forEach(
       metricUrn => {
-        return this._fetchRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
+        return this._cacheRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
       });
+    metricUrns.forEach(
+        metricUrn => {
+          return this._fetchRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
+        });
+  },
+
+
+  /**
+   * Caches the metric data for a row of the metric table
+   *
+   * @param {String} metricUrn Metric urn
+   * @param {Object} context Context
+   * @param {Object} metricUrnToOffestAndUrn Hash map from metric urn to offset and urn
+   * @returns {undefined}
+   */
+  async _cacheRowSlice(metricUrn, context, metricUrnToOffestAndUrn) {
+    const [start, end] = context.anomalyRange;
+    const offsets = metricUrnToOffestAndUrn[metricUrn].map(tuple => tuple[0]);
+    const timezone = moment.tz.guess();
+    const url = `/rootcause/metric/aggregate/cache?urn=${metricUrn}&start=${start}&end=${end}&offsets=${offsets}&timezone=${timezone}`;
+
+    try{
+      fetch(url);
+    } catch (error){
+      // left blank
```

Comment:
If we have plans to handle the error, let's add a TODO comment in the code

Reply:
This should have been awarning that `error` is defined and not used. I rather we just removed this catch.